File: infra/src/main/java/com/beat/infra/notification/slack/vo/message/SlackMessage.java
package com.beat.infra.notification.slack.vo.message;

import com.beat.infra.notification.slack.vo.block.Block;
import java.util.List;

public record SlackMessage(
	List<Attachment> attachments
) {

	public static SlackMessage newInstance(List<Block> blocks, String color) {
		return new SlackMessage(List.of(Attachment.of(color, blocks)));
	}

	public record Attachment(String color, List<Block> blocks) {

		public static Attachment of(String color, List<Block> blocks) {
			return new Attachment(color, blocks);
		}
	}
}

File: infrastructure/src/main/kotlin/com/beat/infra/external/notification/slack/vo/message/SlackMessage.kt
package com.beat.infra.external.notification.slack.vo.message

import com.beat.infra.external.notification.slack.vo.block.Block

internal data class SlackMessage(
    val attachments: List<Attachment>,
) {
    companion object {
        fun newInstance(blocks: List<Block>, color: String): SlackMessage =
            SlackMessage(listOf(Attachment.of(color, blocks)))
    }

    data class Attachment(
        val color: String,
        val blocks: List<Block>,
    ) {
        companion object {
            fun of(color: String, blocks: List<Block>): Attachment = Attachment(color, blocks)
        }
    }
}
